zerver/views/messages.py

```diff
@@ -153,9 +153,7 @@ def _pg_re_escape(self, pattern):
         s = list(pattern)
         for i, c in enumerate(s):
             if c not in self._alphanum:
-                if c == '\000':
```

Comment:
OK, let's go with that plan.  `check_stream_name` in `zerver/lib/actions.py` is the thing to patch.